test.js

```diff
@@ -464,6 +465,26 @@ function runTests(baseopts) {
         });
       });
     });
+    it('should emit two `unlinkDir` event when two nested directories were removed', function(done) {
+      var spy = sinon.spy();
+      var testDir = getFixturePath('subdir');
+      var testDir2 = getFixturePath('subdir/subdir2');
+      var testDir3 = getFixturePath('subdir/subdir2/subdir3');
+      fs.mkdirSync(testDir, 0x1ed);
+      fs.mkdirSync(testDir2, 0x1ed);
+      fs.mkdirSync(testDir3, 0x1ed);
+      watcher.on('unlinkDir', spy).on('ready', function() {
+        w(fse.remove.bind(fse, testDir2, simpleCb))(); // test removing in one
```

Comment:
Can we do this without the added dependency?

It's just a convenience function, doing them both back to back with regular `fs` async calls will have the same effect.

Reply:
Wish it would work without `fs-extra`, but when I test with Node 10.1 I only get the failure with `fs-extra`, but didn't get with two async fs calls in a row.  Apparently something done differently in fs-extra, I didn't look.  Maybe it calls an OS command, I really didn't look under the hood, just don't know its internals.

So could we then say it is fs-extra's "fault"?  No, because same happens if deleting the directory that contains the other directory by user action in Windows Explorer.  So, it just happens fs-extra does whatever is more similar to when a user on the computer uses Windows Explorer to delete.

Reminder the issue we're fixing here is Windows only, apparently Node 10.0 to 10.4 only.  Tested to occur in 10.1 to 10.4, not in 8.11.2, and we now think won't occur in Node 10.4.1, not sure yet.

And the nasty part is the complete crash, exit from app, if a user deletes a directory that contains another directory, e.g. by using Windows Explorer.

Given the severe nastiness of that, use of fs-extra seems justifiable to have a good test case.  Only way known so far to reliably reproduce the defect.

A test that doesn't reproduce the would pass, but it wouldn't really test.

Reply:
Did you try `rimraf`? Apparently that's all that's going on under `fse.remove` - unless they have some modified implementation of that standard dependency that we already use here.

Reply:
You are right, `rimraf` does it as well.  Committed test using `rimraf` instead of `fs-extra`.